web/controllers/search_controller.ex

```diff
@@ -18,8 +18,11 @@ defmodule Constable.SearchController do
     render conn, "new.html", announcements: []
   end
 
-  defp matching_announcements(%{"query" => search_terms}) do
+  defp matching_announcements(params = %{"query" => search_terms}) do
+    excluded_interests = params["exclude_interests"] || []
+
     Announcement.search(search_terms)
+    |> Announcement.without_interests(excluded_interests)
```

Comment:
This should probably be moved to `Announcement.search` since it's always used. So maybe a second arg should be added so it's `Announcement.search(search_terms, exclude_interests: excluded_interests)`